core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala

```diff
@@ -107,8 +107,13 @@ private[spark] class FileShuffleBlockResolver(conf: SparkConf)
     }
   }
 
-  override def getBlockData(blockId: ShuffleBlockId): ManagedBuffer = {
-    val file = blockManager.diskBlockManager.getFile(blockId)
+  override def getBlockData(blockId: ShuffleBlockId, blockManagerId: BlockManagerId)
+    : ManagedBuffer = {
+    val file = if (blockManager.blockManagerId != blockManagerId) {
+      blockManager.diskBlockManager.getShuffleFileBypassNetworkAccess(blockId, blockManagerId)
```

Comment:
This seems like the wrong abstraction. Ideally we should just call `diskBlockManager.getFile(blockId, blockManagerId)` and it will take care of the rest for us. We shouldn't expose the details of how the shuffle files are retrieved here. i.e. I would just do this:
```
override def getBlockData(
    blockId: ShuffleBlockId,
    blockManagerId: BlockManagerId): ManagedBuffer = {
  val file = blockManager.diskBlockManager.getFile(blockId, blockManagerId)
  ...
}
```